### DISCUSSION TITLE
Upgrading to v4: Flowbite issues

### DISCUSSION THREAD

**Discussion by jibiel** [activeadmin/activeadmin#8330] [Help] [answered]
Following [`UPGRADING.md`](https://github.com/activeadmin/activeadmin/blob/master/UPGRADING.md), I'm trying to switch to v4 from vanilla v3 installation and run into an issue with Flowbite.

1. The first exception is:

```ruby
Sprockets::Rails::Helper::AssetNotPrecompiledError at /admin/things

Asset `flowbite/lib/cjs/index.js` was not declared to be precompiled in production.
Declare links to your assets in `app/assets/config/manifest.js`.
  //= link flowbite/lib/cjs/index.js
and restart your server
```

(which is weird because [demo.activeadmin.info](https://github.com/activeadmin/demo.activeadmin.info) doesn't need this)

2. After following the advise from the exception message and adding:

```js
//= link flowbite/lib/cjs/index.js
```

to `app/assets/config/manifest.js` — the new sweet UI is up and running, but the JS plugins don't work and

3. The console has the following exception:

```js
// assets/flowbite/lib/cjs/index.js
Object.defineProperty(exports, "__esModule", { // <= ReferenceError: Can't find variable: exports
    value: true
});
```

4. Which may also be related to the second exception:

```
No route matches [GET] "/assets/flowbite/src/index.ts"
```

Search suggests that `ReferenceError: Can't find variable: exports` may be related to the TS configuration, but TypeScript is not even installed and is not mentioned in the upgrade guide.

I would appreciate any advice on this 🎩

BTW thanks for the great update! 🚀

**Comment by tagliala** [comment, marked as answer]
I had the same issue. In my case, it was because `config/initializers/assets.rb` contained 

```rb
Rails.application.config.assets.paths << Rails.root.join('node_modules')
```

After removing that line, it worked

However, I didn't need that line for my application, so this can break something

**Comment by jibiel** [comment]
@tagliala Thanks! It worked for us too 🙂

I tried to comment out `Rails.application.config.assets.precompile` (which didn't work), but climbed deeper into a rabbit hole and didn't get to that second line.
